### ISSUE TITLE
Uncharted: Golden Abyss™ [PCSA00029]

### ISSUE THREAD

**Issue by Gerold55** [Vita3K/compatibility#1557]
# App summary
- App name: Uncharted: Golden Abyss™
- App serial: PCSA00029
- App version: 1.00

# Vita3K summary
- Version: v0.1.8
- Build number: 3193
- Commit hash: https://github.com/vita3k/vita3k/commit/9837745e
- CPU backend: Dynarmic
- GPU backend: Vulkan

# Test environment summary
- Tested by: Gerold55
- OS: Windows 11
- CPU: Intel Core i5 9300H
- GPU: NVIDIA Geforce GTX 1650
- RAM: 8 GB

# Issues
Graphical issues with the water and occasional frame drops

# Screenshots
![Screenshot (11)](https://github.com/Vita3K/compatibility/assets/13843095/c5e29368-d92d-4737-901d-12050367d400)

# Log

# Recommended labels
- Ingame-

**Comment by DrakeNathanyt** [comment]
Топ

**Comment by 9074abhi** [comment]
Play this game 60 fps ok but flicker issue malig68 mc4 gup this problm

https://github.com/Vita3K/compatibility/assets/97212153/35172150-860c-4e94-8b32-0de15b0d5915

**Comment by Avsynthe** [comment]
> Play this game 60 fps ok but flicker issue malig68 mc4 gup this problm
> 
>  Screen_Recording_20240217_132152.mp4

I get this black texture issue when using OpenGL. Switching to Vulkan solves this issue... mostly

I say mostly because I still get graphical glitches ranging from this black texture (though only on parts of the screen with Vulkan), to crazy coloured textures covering everything like an acid trip.

The fix for this is to not go above 1.75x resolution upscaling, which is an absolute shame on a 4K screen.

**Comment by TadeoCG15** [comment]
One question, did you finally solve it? Because it still happens to me and I did everything it says there, in the comment.

**Comment by benjamintoledo** [comment]
Idk what happened in later updates, but now it just stucks. Move it to "Nothing".

<img width="722" height="431" alt="Image" src="https://github.com/user-attachments/assets/910ee19d-bd35-4f8a-96a6-82cb33903535" />

**Comment by s0unt** [comment]
MacBook Pro M3 Pro
36 GB of memory
The game doesn't render. I tested a lot of different combinations and did not make it work.
<img width="1072" height="684" alt="Image" src="https://github.com/user-attachments/assets/e99cdc02-f54d-473a-92d6-8e26dbf74156" />

**Comment by fpunch1000** [comment]
> MacBook Pro M3 Pro 36 GB of memory The game doesn't render. I tested a lot of different combinations and did not make it work. <img alt="Image" width="1072" height="684" src="https://private-user-images.githubusercontent.com/44710482/599838236-e99cdc02-f54d-473a-92d6-8e26dbf74156.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3ODEyODAzNzIsIm5iZiI6MTc4MTI4MDA3MiwicGF0aCI6Ii80NDcxMDQ4Mi81OTk4MzgyMzYtZTk5Y2RjMDItZjU0ZC00NzNhLTkyZDYtOGUyNmRiZjc0MTU2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjA2MTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwNjEyVDE2MDExMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQxOThmMDBlOGI0ZTE2YzY0NmE1ZTQ4MDY3Mzk1OTY1ZmQ1MDRiZWYxZWQyYmY2ZmNkZGRmYjdhZGFmMzQ2MzQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JnJlc3BvbnNlLWNvbnRlbnQtdHlwZT1pbWFnZSUyRnBuZyJ9.7aIpOX90FvqJ_cz9U--sD6y6VASoZl8AKgr3jmbSErQ">

Same behavior on a MacBook Pro M5 Max 36GB with latest versions.

**Comment by RETRODODO** [comment]
UPDATE: While the saving bug has been fixed, Uncharted on the PS Vita utilizes unique hardware features like gyro controls and the rear light sensor. This makes certain sections impossible to progress past, causing players to get stuck.

Namely:

1- When crossing logs that require balance, gyro controls are required. Some players have managed to bypass this by mapping motion controls to an analog stick via Steam. However, the original reporter noted they only tested this on a Steam Deck, so it is unconfirmed if this workaround functions seamlessly on a standard PC setup.  I personally had to move my saves to my android handheld and the Gyro worked.

2- My current softlock occurs in Chapter 11, where you must hold the back of a parchment up to a bright light. On real hardware, this is triggered by aiming the Vita’s camera/light sensor at a light source. Since Vita3K does not currently emulate this feature, progression is completely blocked here.

-----------------------------------------------------------------------------------------------------------------------------------
PCSA00029 updated to 1.03

The game is now playable with the latest Vita3K update. It is recommended to use Vulkan (OpenGL results in a black screen) with a maximum resolution of 1.75x. It is also advised to update the game to version 1.03 and use the debug menu to unlock all levels as a precaution. Performance averages around 24 FPS at 1.75x resolution, though a stable 30 FPS can be achieved by lowering the resolution.

Any resolution above 1.75x will trigger color glitches.
Visual bugs still persist during water sequences.
The save issue has been fixed.
The glitch causing players to fall into the void while the screen turns gray has been fixed.
The "no gun" glitch has been fixed.
PC Specs: Intel Core i7-7500U CPU / 16GB DDR4 RAM / AMD Radeon 540 (2GB GDDR5)
Attempted to enhance the visuals using ReShade, as the first level is quite dark—but Vita3K fails to launch when ReShade is active.

Tested on Android and it works as well, though it suffers from additional visual artifacts, including black boxes and tiny black graphical glitches during gameplay.

**Comment by xxNoSuRReNDeRxx** [comment]
Seems OK at first no glitches (at 1.75x) anymore plays smooth as butter but it seems to ALWAYS crash on Windows 10 pro Vita 3k when I get to the rocket launcher guy who says "Your a dead man"at the start after climbing and shooting from climbing the first time.... Crashes there every single time - Windows 10 pro - Dual Xeon 5218 golds Quadro Rtx8000(48GBvram) 64gb ddr5 fast NVME ssd

**Comment by RETRODODO** [comment]
> Seems OK at first no glitches (at 1.75x) anymore plays smooth as butter but it seems to ALWAYS crash on Windows 10 pro Vita 3k when I get to the rocket launcher guy who says "Your a dead man"at the start after climbing and shooting from climbing the first time.... Crashes there every single time - Windows 10 pro - Dual Xeon 5218 golds Quadro Rtx8000(48GBvram) 64gb ddr5 fast NVME ssd

You gotta play with your settings until you find the suitable ones that prevent that crash or share the log with the devs. I was able to complete the full game few days ago

**Comment by xxNoSuRReNDeRxx** [comment]
> > Seems OK at first no glitches (at 1.75x) anymore plays smooth as butter but it seems to ALWAYS crash on Windows 10 pro Vita 3k when I get to the rocket launcher guy who says "Your a dead man"at the start after climbing and shooting from climbing the first time.... Crashes there every single time - Windows 10 pro - Dual Xeon 5218 golds Quadro Rtx8000(48GBvram) 64gb ddr5 fast NVME ssd
> 
> You gotta play with your settings until you find the suitable ones that prevent that crash or share the log with the devs. I was able to complete the full game few days ago

I forgot the log... it crashes everytime exactly when he says "see how you like this... "

[vita3k.log](https://github.com/user-attachments/files/29580451/vita3k.log)

**Comment by ErriiickK** [comment]
# App summary
- App name: Uncharted: Golden Abyss
- App serial: PCSA00029
- App version: 1.03 (the same issue also occurs on version 1.00)

# Vita3K summary
- Version: v0.2.1
- Build number: 4058
- Commit hash: https://github.com/Vita3K/Vita3K/commit/6063154f
- GPU backend: Vulkan

# Test environment summary
- OS: Windows 11 64-bit, build 26200
- CPU: AMD Ryzen 5 7600X
- GPU: AMD Radeon RX 9060 XT 16 GB
- RAM: 32 GB DDR5

# Issues

Uncharted: Golden Abyss crashes during the initial boot process on Vita3K
build 4058.
The issue occurs on a clean first boot, without any previously created save
data. I initially tested the base game version 1.00 and later installed update
1.03, but the same crash still occurs.

The game is being tested with the following settings:

- Vulkan renderer
- 1x resolution
- CPU optimisation enabled
- NGS enabled
- Modules mode set to Automatic
- AMD Radeon RX 9060 XT manually selected as the rendering device
- No game patches or graphical modifications enabled

The relevant log section is:

Loading module "vs0:sys/external/libSceNearUtil.suprx"
Module libSceNearUtil is HLE. Skipping load.
Missing file at ".../savedata/PCSA00029/discoverycache.bin"
Import function for NID 0xA412E9CA not found
(thread name: cNearUtility, thread ID: 109)

This is followed by:

Exception EXCEPTION_ACCESS_VIOLATION (0xC0000005)
Write violation at address 0x0

Exception EXCEPTION_ACCESS_VIOLATION (0xC0000005)
Write violation at address 0x0

Because this is the first boot, files such as discoverycache.bin,
bountyattr.bin and the save-slot parameter files have not been created yet.
The emulator continues booting after several of the other missing-save-file
messages, so they do not appear to be the primary cause of the crash.

LLE module test

I also tested changing the module configuration from Automatic to Manual and
forcing libSceNearUtil to load in LLE mode.

With LLE enabled, the original HLE missing-import message changes and the
module is loaded, but the emulator still fails with another unresolved import:

Module SceNearUtil loaded
Import function for NID 0xE426FEE8 not found
Exception EXCEPTION_ACCESS_VIOLATION (0xC0000005)
Write violation at address 0x0

During one test with libSceNearUtil forced to LLE, the game reached the
loading image but remained stuck and did not reach the main menu. Other
attempts still ended with an access violation.

Therefore, forcing the module to LLE does not solve the problem. It only
changes which unresolved Near-related import is reached.

Shader translator warning

The complete log also contains the following shader translator warning:

VNMAD is VDP but has dest mask with more than 1 component active.
Please report to developer to handle this!

However, the emulator continues generating and compiling multiple shaders
after this warning. The Near-related missing import occurs later, immediately
before the access violation, so the connection between this shader warning
and the final crash is unclear.

There are also two unsupported GXM texture address mode errors:

sceGxmTextureSetUAddrModeSafe returned SCE_GXM_ERROR_UNSUPPORTED
sceGxmTextureSetVAddrModeSafe returned SCE_GXM_ERROR_UNSUPPORTED

These occur earlier in the boot process and do not immediately stop the game.

Build comparison

I tested the same game dump and the same computer configuration on multiple
Vita3K builds:

Build 3619: boots successfully and reaches the main menu
Build 4000: crashes during the initial boot
Build 4058: crashes during the initial boot

The game therefore works on build 3619 but fails on builds 4000 and 4058.

This suggests that a regression may have been introduced sometime between
builds 3619 and 4000, possibly involving the HLE implementation of
SceNearUtil, sceNearInitialize, the cNearUtility thread or another
Near-related import.

**Comment by icefox1983** [comment]
> Uncharted: Golden Abyss crashes during the initial boot process on Vita3K build 4058. The issue occurs on a clean first boot, without any previously created save data. I initially tested the base game version 1.00 and later installed update 1.03, but the same crash still occurs.
> 
> The game is being tested with the following settings:
> 
> * Vulkan renderer
> * 1x resolution
> * CPU optimisation enabled
> * NGS enabled
> * Modules mode set to Automatic
> * AMD Radeon RX 9060 XT manually selected as the rendering device
> * No game patches or graphical modifications enabled
> 
> Exception EXCEPTION_ACCESS_VIOLATION (0xC0000005) Write violation at address 0x0
> 
> Exception EXCEPTION_ACCESS_VIOLATION (0xC0000005) Write violation at address 0x0

I had the same crash on my ROG Ally at game launch. It always happened when compiling 13 or 21 shaders when the coin spins once or when the next flash screen shows. My PC with Nvidia GPU runs perfectly. I thus know it's AMD GPU causing it. I further confirmed the fault to be with the AMD Vulkan driver. I downgraded to driver version 32.0.11021.3002 and the game starts fine now, but it crashes randomly now with the same exception 0xC0000005. I was able to downgrade the driver by uninstalling the current driver using DDU and I think Windows automatically installed this version. If you test other versions and find a working driver version let me know, but I suspect none works. AMD's Vulkan driver is notorious. I'll just play on my desktop PC.

**Comment by HylianSoul** [comment]
I have tested [a fork of the Vita3K emulator](https://github.com/nckstwrt/Vita3K-Plus) that solves the problem when you increase the resolution (in my case, is 2x now) and other issues in the game (like random white cubes on the characters eyes when the game is paused).

<img width="962" height="566" alt="Image" src="https://github.com/user-attachments/assets/052d8924-7a96-4872-91d4-fa0c6cd127f3" />

I think it could be an interesting information.

## Device specs

OS: Debian GNU/Linux forky/sid (forky) x86_64
CPU: AMD Ryzen 5 3500U (8) @ 3.70 GHz
GPU: AMD Radeon Vega 8 Graphics [Integrated]
RAM: 32 GiB

The fork, for now, offers only Windows and Android builds. I executed the Windows version using ```proton-cachyos-11.0-20260703-slr-x86_64_v3```.

**Comment by morphinapg** [comment]
> I have tested [a fork of the Vita3K emulator](https://github.com/nckstwrt/Vita3K-Plus) that solves the problem when you increase the resolution (in my case, is 2x now) and other issues in the game (like random white cubes on the characters eyes when the game is paused).
> <img alt="Image" width="962" height="566" src="https://private-user-images.githubusercontent.com/86734055/639756245-052d8924-7a96-4872-91d4-fa0c6cd127f3.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3ODczNjgxNjYsIm5iZiI6MTc4NzM2Nzg2NiwicGF0aCI6Ii84NjczNDA1NS82Mzk3NTYyNDUtMDUyZDg5MjQtN2E5Ni00ODcyLTkxZDQtZmEwYzZjZDEyN2YzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjA4MjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwODIyVDAzMDQyNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWIzNGNhN2JjMDJmOTY3NzNkMWM0YmM2YTkyNWRmNWU5NTJkOTFlMGEzYTgzNmEyZGE2ZmRmNDMxNzgzNDM3MGYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JnJlc3BvbnNlLWNvbnRlbnQtdHlwZT1pbWFnZSUyRnBuZyJ9.TvZs96JoyXXdC_E-smi0ol80p1TQxCv1oPZOIROkbf8">
> 
> I think it could be an interesting information.
> ## Device specs
> 
> OS: Debian GNU/Linux forky/sid (forky) x86_64 CPU: AMD Ryzen 5 3500U (8) @ 3.70 GHz GPU: AMD Radeon Vega 8 Graphics [Integrated] RAM: 32 GiB
> 
> The fork, for now, offers only Windows and Android builds. I executed the Windows version using `proton-cachyos-11.0-20260703-slr-x86_64_v3`.

Can confirm, the game runs at 4K for me with that fork

<img width="3808" height="2159" alt="Image" src="https://github.com/user-attachments/assets/c9e5f1a5-8d47-4c25-88ba-9da44a788457" />

**Comment by ahmedtalha31** [comment]
I heard it had problem with the save is it solved also?

**Comment by HylianSoul** [comment]
> I heard it had problem with the save is it solved also?

Yes. The save data bug was solved in the Vita3K project with [this pull request](https://github.com/Vita3K/Vita3K/pull/4001).

**Comment by xesf** [comment]
Yes, this fork let me run the game wtihout some of the graphical issues on the Steam Machine but brings other issues like not detecting Steam Controller.
